community/kernel/src/main/java/org/neo4j/kernel/impl/util/MultiCloseable.java

```diff
@@ -0,0 +1,104 @@
+ * instances, even if one or more throws {@link IOException} or any other unchecked exception,
+ * before throwing the first exception (or the first IOException, if any, as the parent exception),
+ * with additional exceptions as suppressed.
+ */
+public class MultiCloseable implements Closeable
+{
+    private final Closeable[] closeables;
+
+    public MultiCloseable( Closeable... closeables )
+    {
+        this.closeables = closeables;
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+        Throwable exception = null;
+        for ( Closeable closeable : closeables )
+        {
+            try
+            {
+                closeable.close();
+            }
+            catch ( IOException e )
+            {
+                if ( exception == null )
+                {
+                    exception = e;
+                }
+                else
+                {
+                    if ( !( exception instanceof IOException) )
+                    {
+                        // We end up here if a previous Closeable threw a non-IOException, such that
+                        // the exception variable now points to that exception. In the end, if there are
+                        // only non-IOExceptions thrown it doesn't make sense to convert those into an
+                        // IOException, since it may have unexpected results from the caller POV.
+                        // Therefore we convert the exception variable to an IOException, flipping
+                        // this exception from being suppressed by the existing exception, to being the
+                        // top level exception with the existing one, including all its suppressed,
+                        // as suppressed exceptions of this IOException
+                        Throwable existing = exception;
+                        exception = e;
+
+                        // Unfortunately we cannot flatten out the suppressed hierarchy here so it'll have to be:
+                        // exception: IOException
+                        //   |
+                        //   |-- existing: Throwable
+                        //         |
+                        //         |-- suppressed...: Throwable
+                        exception.addSuppressed( existing );
+                    }
+                    else
+                    {
+                        exception.addSuppressed( e );
+                    }
+                }
+            }
+            catch ( Throwable t )
+            {
+                if ( exception == null )
+                {
+                    exception = t;
+                }
+                else
+                {
+                    exception.addSuppressed( t );
+                }
+            }
+        }
+        if ( exception != null )
+        {
+            throw launderedException( IOException.class, exception );
+        }
+    }
```

Comment:
Why not using `LifeCycle` for this? I don't see the point of adding so much extra complexity since we have already code that does something like this.